src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -105,6 +105,12 @@ void QgsLayoutItemLegend::paint( QPainter *painter, const QStyleOptionGraphicsIt
   if ( !painter )
     return;
 
+  if ( mDeferLegendModelInitialization )
+  {
+    mDeferLegendModelInitialization = false;
+    setCustomLayerTree( mCustomLayerTree.release() );
+  }
```

Comment:
I'd suggest we move this to a private function `ensureModelIsInitialized` , and call that from QgsLayoutItemLegend::model() too.